paasta_itests/steps/paasta_deployd_steps.py

```diff
@@ -29,54 +32,56 @@ def start_deployd(context):
             zk.create('/autoscaling')
         except NodeExistsError:
             pass
-
+    context.zk_hosts = '%s/mesos-testcluster' % get_service_connection_string('zookeeper')
     context.soa_dir = '/nail/etc/services'
     if not hasattr(context, 'daemon'):
-        context.daemon = DeployDaemon()
-        context.daemon.start()
-    for i in range(0, 10):
-        if context.daemon.started:
-            return
-        time.sleep(3)
-    assert context.daemon.started
+        context.daemon = Popen('paasta-deployd', stderr=PIPE)
+    output = context.daemon.stderr.readline()
+    while "Startup finished!" not in output:
+        output = context.daemon.stderr.readline()
+        print(output.rstrip('\n'))
+    time.sleep(5)
 
 
 @then('paasta-deployd can be stopped')
 def stop_deployd(context):
-    context.daemon.stop()
-    for i in range(0, 5):
-        if not context.daemon.is_alive():
-            return
-        time.sleep(3)
-    assert not context.daemon.is_alive()
+    context.daemon.terminate()
+    context.daemon.wait()
 
 
 @then('a second deployd does not become leader')
 def start_second_deployd(context):
-    context.daemon1 = DeployDaemon()
-    context.daemon1.start()
+    context.daemon1 = Popen('paasta-deployd', stderr=PIPE)
+    output = context.daemon1.stderr.readline()
+    fd = context.daemon1.stderr
+    fl = fcntl.fcntl(fd, fcntl.F_GETFL)
+    fcntl.fcntl(fd, fcntl.F_SETFL, fl | os.O_NONBLOCK)
```

Comment:
Sure. The summary is this is some sort of linux hack to make reading the stderr file non-blocking. That way we can keep reading it, waiting for a log line and still have a timeout if nothing gets written to stderr.